### DISCUSSION TITLE
How Many Zigbee Devices Can One Coordinator Hold?

### DISCUSSION THREAD

**Discussion by LordNex** [tube0013/tube_gateways#171] [Q&A]
I currently have the 2023 EFR32 MGM21 with about 24 devices connected. Most stay pretty steady except for a few that are on the edge of reach or has too much interference. As I add more and more of these this will be pretty important. Run off a well and pump so I have leak sensors at various parts of my house that if they detect water, triggers a Zigbee mechanical servo that pulls the main water valve shut. So I guess you could put my Zigbee into 2 categories. Things that are super important like I just described, and other things that are not as important such as temp and humidity sensors and the like. 

So my over all questions is, how many Zigbee devices can you have on one coordinator?

I know you can't run to coordinators in the same zone, but would it be plausible to have one coordinator for cannot fail devices and another for daily functions? Or should I just look at buying more repeaters like those made by Aqara?

Thanks in advance and your product rocks! I can't wait to see what matter brings to the table once it's fully out.

**Comment by LordNex** [comment]
My current topology which is constantly changing. 

![image](https://github.com/tube0013/tube_gateways/assets/9270948/8e12e91d-adde-4b80-ad5b-39425985bc3b)

**Reply by LordNex** [reply]
For the EFR32 MGM21 PoE I bought. Is that the best chipset for a large network? Or would something else be different. If figured the PoE so I can get the antenna into the attic of my house to get the best coverage. 

I originally did a conversion using a backup file but I kept getting small end battery driven devices fall off the network. And some are not in easy to reach locations. So I finally decided for the time being to use the indoor 2.4 ghz radio from a cell repeater, and put it on a Class 4 PoE switch so it would have more power if it needed it. Currently it's barely pulling about 1.7watts which is nothing compared to my WAPs suck up. Luckily my PoE switch will handle up to 124 Watts. And then rebuilt the mesh from scratch. Removed ZHA completely, readded and configured the coordinator being careful not to select my HUBZ1 that's running ZWave, and repaired everything. So I Hoping this solves the problem. I have about 8-10 mains powered LEDVANCE bulbs that show up as routers and allow you to add devices directly from them but they are pushing it with more than 4-6 connections. 

How many connections can that device handle on one network? I have a garage about 50-75 feet from my house which is going to need Zigbee as well. Am I going to need to buy a separate coordinator for that area or is there a way to make it reach?

**Reply by tube0013** [reply]
I have over 150 devices connected to my mesh with an efr32 coordinator. The network size really depends on the number of routers as each add network capacity for more end devices. The current fw for the efr32 supports 32 direct child devices. Most routers can support 5 or more end devices, and any number of other connections to a router.

I spent some down time at the office last week moving things around in my mesh visualization to show the coordinator surrounded by the routers directly connecting to it surrounded by the router connected to the first tier of router and then grouped the end devices around the router they were communicating with. 

![image](https://github.com/tube0013/tube_gateways/assets/11084412/967f7717-1c4d-4324-acb2-df7b9dac9b59)


![image](https://github.com/tube0013/tube_gateways/assets/11084412/e16c2617-45b2-495e-a532-505645eebc73)


![image](https://github.com/tube0013/tube_gateways/assets/11084412/4ade4b1a-c0e8-4940-b58c-d04f547b75e0)

![image](https://github.com/tube0013/tube_gateways/assets/11084412/d2ab540b-0a3f-40a6-a8af-ce81b3241e18)

With 50 routing devices you can get up to a minimum of 250 additional end device connections. Hi

**Reply by LordNex** [reply]
> I have over 150 devices connected to my mesh with an efr32 coordinator. The network size really depends on the number of routers as each add network capacity for more end devices. The current fw for the efr32 supports 32 direct child devices. Most routers can support 5 or more end devices, and any number of other connections to a router.
> 
> I spent some down time at the office last week moving things around in my mesh visualization to show the coordinator surrounded by the routers directly connecting to it surrounded by the router connected to the first tier of router and then grouped the end devices around the router they were communicating with. 
> 
> ![image](https://github.com/tube0013/tube_gateways/assets/11084412/967f7717-1c4d-4324-acb2-df7b9dac9b59)
> 
> 
> ![image](https://github.com/tube0013/tube_gateways/assets/11084412/e16c2617-45b2-495e-a532-505645eebc73)
> 
> 
> ![image](https://github.com/tube0013/tube_gateways/assets/11084412/4ade4b1a-c0e8-4940-b58c-d04f547b75e0)
> 
> ![image](https://github.com/tube0013/tube_gateways/assets/11084412/d2ab540b-0a3f-40a6-a8af-ce81b3241e18)
> 
> With 50 routing devices you can get up to a minimum of 250 additional end device connections. Hi

For one, I have no idea how your able to control your mesh in the way you have. It's really amazing and cool as hell. Your doing what I'm trying to do, but be adding them slowly. At first I didn't fully understand the protocols. I do Microsoft networking for a living so most of my time is spend in AD, Azure, and Group policies. I'm also highly versed in VLAN setup and configurestions. I just finished a project down in the small island country of Nevis where this estate covered about half of an old volcanic ruins. I had to install 75 Wireless access points with 10 separate VLANS with a huge infristrcture. 

I ended up picking up this

Aeotec Range Extender Zi - Zigbee Network Extender - Improves Range and Stability of Zigbee Network https://a.co/d/hlFSeh3

Which is based on an EFR32MG21 chipset and it seems to have resolved slot of problems. I wanted to buy your extender but it was just to expensive. I have a bout 8 more bulbs coming, most are ledvance so they are repeaters. 

How did you get your layout to build that way? Everytime I go to that page its a completely different topology. Now I understand how meshing works and now I should have a strong enough signal to do something similar. Currently I'm St 29 devices and it just keeps increasing. My plan is to replace every bulb I can with a repeatable bulb. But the mesh seems to be in a constant state of flux. 

@tube0013 im not like probably a lot of the people you sell too. I tend to hyper fixate on a problem until its resolved. Here is my current setup and layout. Of course I can goto s different page and back and it's changed again. I'm playing to build around the coordinator first with routable mains powered devices. But that Aeotec I bought seems to be doing the trick. I haven't had anything drop yet.

**Reply by LordNex** [reply]
![image](https://github.com/tube0013/tube_gateways/assets/9270948/d4ee8c3a-52ad-4ee3-ab63-c3025c360317)

![image](https://github.com/tube0013/tube_gateways/assets/9270948/97c0e253-18e6-406f-90c7-775d6e1248b2)

**Comment by MattWestb** [comment]
Then working in Zogbee 3 mode (older is 64k devices) its need storage for updated TC-link keys in the coordinator and its depends of its firmware and hardware configuration and normally its some hundreds.
For EZSP in ZHA its little different then using hashed TC-link keys (using the device IEEE and hashing one key of it and dont need storage for every device only the seed of it) with full security and the max is 0xffff = 64k addresses in the network (minus some reserved addresses like ).
The key for getting the mesh network working good is good routers as one back bone and its working well (only 80 devices in EZSP production network but have some RCP test networks with around 50 test devices) and its very autonomous (bonded light devices so its working if the coordinator is offline) and working well (only Philips HUE motion and Aqara sensors is little tricky).

Edit: I have blocking the coordinator for having direct children and its working better and some good IKEA routers is having around 10 children and working well and the end devices is not going offline or getting problems then the coordinator is off line.

**Reply by LordNex** [reply]
That's the most and best explanation yet. I'm deep into IEEE standards being A+ and working on meter plus certifications. I have 25 years of on hand IT experience primary in Microsoft Enterprise networking. So things such as Active Directory and the myriad of products MS puts out. , this year they put an 11 on the box! And moved the start. Button again. 

Anyway, after buying TubeZ EFR32 based PoE I still wasn't handing the greatest results. My server and coordinator are tword one end of my house yet I have some important end devices on the other side. I added this little daddy

Aeotec Range Extender Zi - Zigbee Network Extender - Improves Range and Stability of Zigbee Network https://a.co/d/izV1Jax

To the mix and so far it's solved my issues. It's also based on the EFR32MG21 chipset so I figured it would mix well. I would have like to gotten TubeZ extender but @ $75+ US I just couldn't warrant the extra cost without any verification or anything of how well it will work. I have a better external extender coming so when that happens the TubesZ is going in the attic where it can have a more equal chance to reach the devices in my wood framed home with a but load of IT gear. 

The trick I haven't figured out is what he shows in some of his screenshots of this huge network. Mine never stays this homogenous and it's constantly changing its mind on what the best topology is. I've tried pairing specific end points for specific bulb/routers but they end up going back to whatever they want to do. I noticed when I added the new repeater today it did make its own Zigbee Group call Linked Lights. I was unsure weather to put all of my routers attached to it and there was nothing in the manual or anything I found on the net to explain it. So this does make a lot of sense

Could you explain how you went about getting yours setup? I've completely wiped the network clean before and just started with the coordinator. I then in the best spiral I could get, added each bulb or device that was considered a router to the coordinator. After that I just let Zigbee decide on what device to add it too by just going to the integration and clicking add device instead of going to a specific router and adding from there. I still have a few more bulbs coming that are ledvance routers I got free cause they screwed up and sent me the "more expensive BT ones from SYLVANIA LEDVANCE. I have no idea how to integrate those except adding them to HomeKit and hoping they make thier way back to HA. 

I'm truly thankful for you taking the time to explain further. And I think I'm adding some back bone. But take a look as it's now with that $30 repeater and it's made a big difference and supposed to reach up to 100 meters line of sight. 

![image](https://github.com/tube0013/tube_gateways/assets/9270948/3de6b209-fca3-4034-b898-94ee9b4ef571)
